db/migrate/20220922190819_create_target_matcher_instances.rb

```diff
@@ -0,0 +1,11 @@
+class CreateTargetMatcherInstances < ActiveRecord::Migration[7.0]
+  def change
+    create_table :target_matcher_instances do |t|
+      t.belongs_to :user
+      t.belogns_to :target
+      t.belogns_to :target_matcher
```

Comment:
As them are keys, we should avoid null values. 
```suggestion
      t.belongs_to :user, null: false
      t.belogns_to :target, null: false
      t.belogns_to :target_matcher, null: false
```